Recommend_For_Student.R

```diff
@@ -1,9 +1,49 @@
-Find similar students (Neighbors)= weighted average of rankings of items they rated, sort.  IF completed, then move to second completed
+# Load the CSV Files
 
-T2.SIM <- cosine(t_comb_matrix)
-diag(T2.SIM) <- NA
-head(rownames(T2.SIM[order(T2.SIM[last.unit,], decreasing = TRUE),]), n = 3)
+I1 <- read.csv("interest.csv", header=TRUE)
+D1 <- read.csv("difficulty.csv", header =TRUE)
 
-important_people <- rbind(t_comb_matrix[,"mh3736"],t_comb_matrix[,"jl4965"], t_comb_matrix[,"ma3694"])
-important_people[,1] <- important_people[,1] * .3399770
-order_values <- as.matrix(rowMeans(important_people))
\ No newline at end of file
+#intialize the variables and build first set of matricies 
+my.uni = "js4633"
+i1names <- I1$stid
+d1names <- D1$stid
+I2 <-data.matrix(I1 [,-1])
+D2 <- data.matrix(D1[,-1])
+rownames(I2) <- i1names
+rownames(D2) <- d1names
+
+#combine difficulty and interest into single 2 column matrix
+two_column <- rbind(cbind(I2[,1],D2[,1]), cbind(I2[,2],D2[,2]), cbind(I2[,3],D2[,3]), cbind(I2[,4],D2[,4]),cbind(I2[,5],D2[,5]),cbind(I2[,6],D2[,6]))
+
+#Perform PCA on new matrix
+tc_pca <- prcomp(two_column)
+
+#Take the primary component from the PCA and rebuild 6 column matrix with one column per module
+combined_matrix <- cbind(tc_pca[["x"]][1:22,1], 
```

Comment:
This is the exact way a programmer would do this manipulation! Not bad, just not very R :) I would take a look at the [tidyr/dplyr cheat sheet](https://www.rstudio.com/wp-content/uploads/2015/02/data-wrangling-cheatsheet.pdf) to get a handle on *one* way of doing these operations. But as with everything in R, there are 50 different ways to do one thing. It may be worth also checking out some reading on [vectorization in R](http://www.noamross.net/blog/2014/4/16/vectorization-in-r--why.html) just to get the idea of where the designers are coming from.